### PR TITLE
migration fix

### PR DESCRIPTION
**Pull request by akshaydeo**
## Summary

Fixes a base64 decode crash in the database migration that drops deployment columns and adds aliases. The issue occurred when encrypted rows contained plaintext deployment JSON data, causing the migration to fail when attempting to decrypt already-plaintext values.

## Changes

- Added encryption fixup logic in `migrationDropDeploymentColumnsAndAddAliases` to properly handle mixed encryption states
- The migration now checks if `aliases_json` values are already encrypted before attempting to encrypt them
- Added comprehensive test coverage for the encryption edge case with `TestMigrationDropDeploymentColumnsAndAddAliases_EncryptedRows`
- Test includes setup helper `setupKeyDBWithLegacyDeploymentColumns` to simulate pre-migration database state

## Type of change

- [x] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Chore/CI

## Affected areas

- [x] Core (Go)
- [ ] Transports (HTTP)
- [ ] Providers/Integrations
- [ ] Plugins
- [ ] UI (Next.js)
- [ ] Docs

## How to test

Run the migration tests to verify the fix:

```sh
go test ./framework/configstore -run TestMigrationDropDeploymentColumnsAndAddAliases_EncryptedRows -v
```

Test the broader migration functionality:

```sh
go test ./framework/configstore -run TestMigration -v
```

Ensure encryption is enabled when running tests that depend on it.

## Screenshots/Recordings

N/A

## Breaking changes

- [ ] Yes
- [x] No

## Related issues

Fixes migration crashes when processing encrypted rows with plaintext deployment data during the aliases migration.

## Security considerations

This change improves the security posture by ensuring consistent encryption of sensitive deployment data during migrations. The fix prevents potential data corruption or exposure that could occur from failed migrations.

## Checklist

- [x] I read `docs/contributing/README.md` and followed the guidelines
- [x] I added/updated tests where appropriate
- [ ] I updated documentation where needed
- [x] I verified builds succeed (Go and UI)
- [x] I verified the CI pipeline passes locally if applicable

### CODE REVIEW

**Comment by akshaydeo** [comment]
* **#2579** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/2579?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/maximhq/bifrost/2579?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `v1.5.0`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: framework/configstore/migrations_test.go [hunk 1]
```diff
@@ -884,3 +885,208 @@ func TestMigrationAddStoreRawRequestResponseColumn_Idempotent(t *testing.T) {
...
+}
+
+// TestMigrationDropDeploymentColumnsAndAddAliases_EncryptedRows tests that the
+// aliases migration correctly handles rows where encryption_status is 'encrypted'
+// but deployment columns contain plaintext JSON (because those columns were never
+// in the encryption list). This was the root cause of the base64 decode crash.
+func TestMigrationDropDeploymentColumnsAndAddAliases_EncryptedRows(t *testing.T) {
+	if !encrypt.IsEnabled() {
+		t.Skip("encryption not enabled, skipping encrypted rows test")
+	}
```
[the remark this answers is not in the document]
Comment (akshaydeo) [reply framework/configstore/migrations_test.go:971]: This is already deterministic — `encryption_test.go` in the same package has an `init()` that always calls `encrypt.Init(testEncryptionKey, ...)`. Since Go compiles all `_test.go` files in a package together, this `init()` runs before any test in the `configstore` package, so `encrypt.IsEnabled()` is always `true`. The `t.Skip` is just a defensive safety net, not a CI gap.

**Comment by akshaydeo** [comment]
### Merge activity

* **Apr 10, 7:30 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/2579).
* **Apr 10, 7:31 AM UTC**: @akshaydeo merged this pull request with [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/2579).
